Update Linux perfcollect docs to disable Write-No-Execute (#32338)
dotnet/docs: docs/core/diagnostics/diagnostics-in-containers.md

  - `DOTNET_PerfMapEnabled=1`
  - `DOTNET_EnableEventLog=1`

> [!NOTE]

> > When executing the app with .NET 7, please set `DOTNET_EnableWriteXorExecute=0` in addition to the above environment variables.

Reviewer: ```suggestion
> When executing the app with .NET 7, you must also set `DOTNET_EnableWriteXorExecute=0` in addition to the preceding environment variables.
```
